### PR TITLE
Upgrade prometheus library

### PR DESCRIPTION
**Pull request by ChrsMark**
Copy of https://github.com/elastic/beats/pull/28023

## What does this PR do?
This PR updates the `github.com/prometheus/prometheus` and makes the proper LICENCE overrides for the dependencies:

```
go get github.com/prometheus/prometheus@v1.8.2-0.20210701133801-b0944590a1c9
make notice
```

This is a prerequisite for https://github.com/elastic/beats/pull/27269

~~Detailed licence checks for libs:~~
~~1. `github.com/ajstarks/svgo` -> [CC-BY-4.0](https://github.com/ajstarks/svgo/blob/master/LICENSE)~~
~~2. `github.com/bmizerany/pat` -> from README.md~~
~~3. `github.com/cactus/go-statsd-client/statsd` -> [MIT](https://github.com/cactus/go-statsd-client/blob/master/LICENSE.md)~~
~~4. `github.com/dnaeon/go-vcr` -> [BSD-2-Clause](https://github.com/dnaeon/go-vcr/blob/master/LICENSE)~~
~~5. `github.com/golang/freetype` -> [GPL-2.0](https://github.com/golang/freetype/blob/master/licenses/gpl.txt)~~
~~6. `github.com/hudl/fargo` -> [MIT] from README.md~~
~~7. `github.com/influxdata/line-protocol` -> [MIT](https://github.com/influxdata/line-protocol/blob/master/LICENSE)~~
~~8. `github.com/lightstep/lightstep-tracer-common/golang/gogo` -> [MIT](https://github.com/lightstep/lightstep-tracer-common/blob/master/LICENSE)~~
~~9. `github.com/pascaldekloe/goe` -> [CC0-1.0](https://github.com/pascaldekloe/goe/blob/master/LICENSE)~~
~~10. `github.com/streadway/handy` -> [BSD 2-Clause ] from README.md~~
~~11. `gonum.org/v1/plot` -> [BSD-3-Clause](https://github.com/gonum/license/blob/master/LICENSE)~~

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -30,7 +30,7 @@ require (
 	github.com/apoydence/eachers v0.0.0-20181020210610-23942921fe77 // indirect
 	github.com/armon/go-socks5 v0.0.0-20160902184237-e75332964ef5
 	github.com/aws/aws-lambda-go v1.13.3
-	github.com/aws/aws-sdk-go v1.19.48
+	github.com/aws/aws-sdk-go v1.38.60
```
Comment (ruflin) [inline go.mod:33]: @kaiyan-sheng As a follow up, would be good to figure out why we have 2 aws sdk ..

**Review by ruflin — APPROVED** [review]
I'm good moving forward with this. Somehow CI has a complaint.

**Comment by ChrsMark** [comment]
Update here. No need for the overrides in `overrides.json`, this was an issue that I had locally because of the `GOFLAGS="-mod=mod"` that was set on my end.  It was resolved under other scope discussing with @jsoriano :).

Also I remove the  replace for docker since this PR will hit 8.0.

**Review by jsoriano — APPROVED** [review]
:+1:

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -27,6 +27,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Enable IMDSv2 support for `add_cloud_metadata` processor on AWS. {issue}22101[22101] {pull}28285[28285]
 - Update kubernetes.namespace from keyword to group field and add name, labels, annotations, uuid as its fields {pull}27917[27917]
 - Previously, RE2 and thus Golang had a bug where `(|a)*` matched more characters than `(|a)+`. To stay consistent with PCRE, the bug was fixed. Configurations that rely on the old, buggy behaviour has to be adjusted. See more about Golang bug: https://github.com/golang/go/issues/46123 {pull}27543[27543]
+- Remove replace for docker library. {pull}28716[28716]
```
Comment (jsoriano) [inline CHANGELOG.next.asciidoc:30]: This is not a user facing change :slightly_smiling_face: In any case we could say something as `Update docker client`.

**Comment by jsoriano** [comment]
@ChrsMark it seems this change is not in 8.0 branch, can you please backport it if needed?

**Comment by jsoriano** [comment]
Ok, seen in an email thread that this is not needed in 8.0 branch.

**Pull request merged**